### PR TITLE
Diagonal tensor methods

### PR DESCRIPTION
**Pull request by TheBB**
Just for convenience.

### CODE REVIEW

File: src/Utility/Tensor.C [hunk 1]
```diff
@@ -255,6 +255,22 @@ Tensor::Tensor (const std::vector<Real>& a, bool transpose) : n(sqrt(a.size()))
...
 
+void Tensor::diag(double value)
+{
+  this->zero();
+  for (t_ind i = 1; i <= n; i++)
+    (*this)(i,i) = value;
+}
+
+
+void Tensor::diag(Vec3 diagonal)
```
Comment (akva2) [inline src/Utility/Tensor.C]: no need to copy.

**Comment by TheBB** [comment]
Updated

File: src/Utility/Tensor.C [hunk 2]
```diff
@@ -255,6 +255,22 @@ Tensor::Tensor (const std::vector<Real>& a, bool transpose) : n(sqrt(a.size()))
...
 
+void Tensor::diag(double value)
+{
+  this->zero();
+  for (t_ind i = 1; i <= n; i++)
+    (*this)(i,i) = value;
+}
+
+
+void Tensor::diag(Vec3& diagonal)
```
Comment (akva2) [inline src/Utility/Tensor.C]: getting there. now make it immutable and i'll merge ;)

Comment (TheBB) [inline src/Utility/Tensor.C]: :+1:

**Comment by akva2** [comment]
merged manually.

**Pull request closed without merging**